check_diff/src/main.rs

```diff
@@ -0,0 +1,20 @@
+use clap::Parser;
+/// Inputs for the check_diff script
+#[derive(Parser)]
+struct CliInputs {
+    /// URL to remote repo
+    remote_repo_url: String,
+    /// The path to the file to read
+    feature_branch: String,
+    optional_commit_hash: Option<String>,
+    #[clap(value_delimiter = ',', num_args = 1..)]
+    optional_rustfmt_config: Option<Vec<String>>,
```

Comment:
Actually now that I think about it, we might want to also add doc comments for the `optional_commit_hash` and the `optional_rustfmt_config`, and while we're at it I think we can just call the fields `commit_hash` and `rustfmt_configs`. The `Option<T>` should tell us that it's optional.

It might also be useful to derive long and short names for these options

Reply:
<img width="1046" alt="Screenshot 2024-05-30 at 4 12 58 PM" src="https://github.com/rust-lang/rustfmt/assets/67570985/3efb583a-3758-4519-81ac-25318cf913fd">
After adding the long and short names for optional parameters, this is the sample usage. It can be quite misleading for users since the usage appears to be 
```
check_diff -c commit-hash <REMOTE_REPO_URL> <FEATURE_BRANCH>
```
This usage is valid though it differs from the original script written.

Reply:
@benluiwj that's fine. At the end of the day this is a tool for rustfmt CI so I don't mind if we can now specify arguments in a different order than the original script, but it's good that you're highlighting that point.